src/IonEvent.ts

```diff
@@ -0,0 +1,473 @@
+            writer.writeInt(binaryBuffer[i]);
+        }
+        writer.endContainer();
+    }
+
+    equals(expected : IonEvent) : boolean {
+        return (
+            this.eventType === expected.eventType &&
+            this.ionType === expected.ionType &&
+            this.depth === expected.depth &&
+            this.annotationEquals(expected.annotations) &&
+            this.valueEquals(expected)
+        );
+    }
+
+    annotationEquals(expectedAnnotations : string[]) : boolean {
+        if(this.annotations.length !== expectedAnnotations.length) return false;
+        for(let i = 0; i < this.annotations.length; i++){
+            if(this.annotations[i] !== expectedAnnotations[i]) return false;
+        }
+        return true;
+    }
+}
+
+export class IonEventFactory {
+    makeScalarEvent(ionType : IonType, fieldName : string, depth : number, annotations : string[], reader : Reader) : IonEvent {
+        let eventType = IonEventType.SCALAR;
+
+        if (reader.isNull()) return new IonNullEvent(eventType , ionType, fieldName, annotations, depth);
+
+        switch(ionType) {
+            case IonTypes.BOOL : {
+                return new IonBoolEvent(eventType , ionType, fieldName, annotations, depth, reader.value());
+            }
+            case IonTypes.INT : {
+                return new IonIntEvent(eventType , ionType, fieldName, annotations, depth, reader.value());
+            }
+            case IonTypes.FLOAT : {
+                return new IonFloatEvent(eventType , ionType, fieldName, annotations, depth, reader.value());
+            }
+            case IonTypes.DECIMAL : {
+                return new IonDecimalEvent(eventType , ionType, fieldName, annotations, depth, reader.value());
+            }
+            case IonTypes.SYMBOL : {
+                return new IonSymbolEvent(eventType , ionType, fieldName, annotations, depth, reader.value());
+            }
+            case IonTypes.STRING : {
+                return new IonStringEvent(eventType , ionType, fieldName, annotations, depth, reader.value());
+            }
+            case IonTypes.TIMESTAMP : {
+                return new IonTimestampEvent(eventType , ionType, fieldName, annotations, depth, reader.value());
+            }
+            case IonTypes.BLOB : {
+                return new IonBlobEvent(eventType , ionType, fieldName, annotations, depth, reader.value());
+            }
+            case IonTypes.CLOB : {
+                return new IonClobEvent(eventType , ionType, fieldName, annotations, depth, reader.value());
+            }
+
+            default : {
+                throw new Error("IonType " + ionType.name + " unexpected.");
+            }
+        }
+    }
+
+    makeContainerEvent(eventType : IonEventType, ionType : IonType, fieldName : string, annotations : string[], depth : number, ionValue : any) : IonContainerEvent {
+        switch(eventType) {
+            case IonEventType.CONTAINER_START : {
```

Comment:
Is there a reason to have this outer `switch` if you are only checking to see if it is `IonEventType.CONTAINER_START`? Why not an if/else?